addons/website_event/tests/test_website_event.py

```diff
@@ -58,6 +59,41 @@ def test_canonical_url(self):
 
 @tagged('post_install', '-at_install')
 class TestWebsiteAccess(HttpCaseWithUserDemo, TestWebsiteEventCommon):
+    @staticmethod
+    def are_names_in_order(names_in_expected_order, text):
+        last_index = -1
+        if isinstance(text, bytes):
+            text = text.decode('utf-8')
+        for name in names_in_expected_order:
+            current_index = text.find(name, last_index + 1)
+            if current_index == -1:
+                if text.find(name) != -1:
+                    return False
+                else:
+                    raise Exception(f"I can't check ordering when name: '{name}' is not present in the text.")
+            elif current_index <= last_index:
+                return False
+            last_index = current_index
+        return True
```

Comment:
This isn't specific enough, there's no indication of what elements on the page we're trying to match on.

I would use lxml and use xpath to search for the elements properly with
```py
self.assertListEqual(sorted_events.mapped('name'), [element.text for element in lxml.html.fromstring(res.content).findall(xpath_for_event_title_cards)]
```

And then probably just inline it inside the test.

Or maybe even better, just patch request.render and check the order of the values directly without going through qweb.